### PR TITLE
fixed rights retention strategy data model and removed some code styl…

### PR DESCRIPTION
**Pull request by ketilaa**
…e warnings

### CODE REVIEW

File: nva-datamodel-java/src/main/java/no/unit/nva/model/associatedartifacts/AbstractRightsRetentionStrategy.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import static java.util.Objects.nonNull;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Objects;
+
+/**
+ * The AbstractRightsRetentionStrategy abstract class represents the strategy for rights retention. It makes use of
+ * Jackson annotations to handle JSON serialization and deserialization into the correct implementing types:
+ * CustomerRightsRetentionStrategy, OverriddenRightsRetentionStrategy and NullRightsRetentionStrategy.
+ */
+/* package visible (*/ abstract class AbstractRightsRetentionStrategy implements RightsRetentionStrategy {
```
Comment (brinxmat) [inline nva-datamodel-java/src/main/java/no/unit/nva/model/associatedartifacts/AbstractRightsRetentionStrategy.java:12]: The "(" is probably a typo here? Is this equivalent to `@VisibleForTesting`?

File: nva-datamodel-java/src/test/java/no/unit/nva/PublicationMapperTest.java [hunk 2]
```diff
@@ -21,18 +21,16 @@
...
     public static final ObjectNode SOME_CONTEXT = dataModelObjectMapper.createObjectNode();
 
     @Test
     void canMapPublicationAndContextToPublicationResponse() throws Exception {
-
         Publication publication = PublicationGenerator.randomPublication();
         var json = JsonUtils.dtoObjectMapper.writeValueAsString(publication);
-        var des = JsonUtils.dtoObjectMapper.readValue(json,Publication.class);
-        assertThat(des,doesNotHaveEmptyValuesIgnoringFields(Set.of("doiRequest", BOOK_REVISION_FIELD)));
+        var des = JsonUtils.dtoObjectMapper.readValue(json, Publication.class);
```
Comment (brinxmat) [inline nva-datamodel-java/src/test/java/no/unit/nva/PublicationMapperTest.java:30]: What's `des`?

File: nva-datamodel-java/src/test/java/no/unit/nva/model/file/FileModelTest.java [hunk 3]
```diff
@@ -289,7 +290,7 @@ private File getPublishedFileWithCustomRRS(UUID identifier) {
             .withName(FIRST_FILE_TXT)
             .withPublisherAuthority(false)
             .withSize(SIZE)
-            .withRightsRetentionStrategy(CustomerRightsRetentionStrategy.create())
+            .withRightsRetentionStrategy(CustomerRightsRetentionStrategy.create(UNKNOWN))
```
Comment (brinxmat) [inline nva-datamodel-java/src/test/java/no/unit/nva/model/file/FileModelTest.java:293]: Or potentially Random element…

File: nva-datamodel-testutils/src/main/java/no/unit/nva/model/testing/associatedartifacts/util/RightsRetentionStrategyGenerator.java [hunk 4]
```diff
@@ -1,28 +1,33 @@
...
     public static RightsRetentionStrategy randomRightsRetentionStrategy() {
         RightsRetentionStrategy[] strategies = {
-            CustomerRightsRetentionStrategy.create(),
-            FunderRightsRetentionStrategy.create(),
-            OverriddenRightsRetentionStrategy.create(),
-            NullRightsRetentionStrategy.create()
+            CustomerRightsRetentionStrategy.create(UNKNOWN),
+            FunderRightsRetentionStrategy.create(UNKNOWN),
+            OverriddenRightsRetentionStrategy.create(UNKNOWN, randomString()),
+            NullRightsRetentionStrategy.create(UNKNOWN)
```
Comment (brinxmat) [inline nva-datamodel-testutils/src/main/java/no/unit/nva/model/testing/associatedartifacts/util/RightsRetentionStrategyGenerator.java:30]: Ok, maybe you choose unknown for a reason

File: nva-datamodel-java/src/test/java/no/unit/nva/model/file/FileModelTest.java (hunk 3 above)
Comment (ketilaa) [reply nva-datamodel-java/src/test/java/no/unit/nva/model/file/FileModelTest.java:293]: There will be some rules enforced as to what are valid combinations.Not sure if I want to generate random elements that does not make sense here. Clients not setting the field should end up with DEFAULT until the object has been through the business logic and finally stored with a "valid" value in the database.

File: nva-datamodel-testutils/src/main/java/no/unit/nva/model/testing/associatedartifacts/util/RightsRetentionStrategyGenerator.java (hunk 4 above)
Comment (ketilaa) [reply nva-datamodel-testutils/src/main/java/no/unit/nva/model/testing/associatedartifacts/util/RightsRetentionStrategyGenerator.java:30]: Might adjust this when I see how it turns out when using it in the publication-api.

File: nva-datamodel-java/src/test/java/no/unit/nva/PublicationMapperTest.java (hunk 2 above)
Comment (ketilaa) [reply nva-datamodel-java/src/test/java/no/unit/nva/PublicationMapperTest.java]: `deserializedPublication` ✅

**Review by brinxmat — APPROVED** [review]

**Pull request merged**